make/autoconf/jdk-version.m4

```diff
@@ -109,6 +109,33 @@ AC_DEFUN_ONCE([JDKVER_SETUP_JDK_VERSION_NUMBERS],
   fi
   AC_SUBST(MACOSX_BUNDLE_ID_BASE)
 
+  # The launcher name, if any
+  AC_ARG_WITH(launcher-name, [AS_HELP_STRING([--with-launcher-name],
+      [Set launcher name. Part of the -version and -fullversion output. @<:@not specified@:>@])])
+  if test "x$with_launcher_name" = xyes; then
+    AC_MSG_ERROR([--with-launcher-name must have a value])
+  elif [ ! [[ $with_launcher_name =~ ^[[:print:]]*$ ]] ]; then
+    AC_MSG_ERROR([--with-launcher-name contains non-printing characters: $with_launcher_name])
+  elif test "x$with_launcher_name" != x; then
```

Comment:
This should be
```
  elif test "x$with_launcher_name" != x -a "x$with_launcher_name" != xno; then
```
to properly handle `--without-launcher-name`. Same goes below to handle `--without-product-name`.